src/rust/engine/process_execution/src/named_caches.rs

```diff
@@ -51,13 +55,27 @@ impl NamedCaches {
   pub fn local_paths<'a>(
     &'a self,
     caches: &'a BTreeMap<CacheName, RelativePath>,
-  ) -> impl Iterator<Item = WorkdirSymlink> + 'a {
-    caches
+  ) -> Result<Vec<WorkdirSymlink>, String> {
+    let symlinks = caches
       .iter()
       .map(move |(cache_name, workdir_rel_path)| WorkdirSymlink {
         src: workdir_rel_path.clone(),
         dst: self.local_base.join(&cache_name.0),
       })
+      .collect::<Vec<_>>();
```

Comment:
This change has been extracted to https://github.com/pantsbuild/pants/pull/16685.